Add dependency management, Webpack, ES6 and so on.

daeyeon opened the pull request:
Migrating the sources under /www to the one based on webpack and ES6.

IoT.jsCode-DCO-1.0-Signed-off-by: Daeyeon Jeong daeyeon.jeong@samsung.com

daeyeon commented:
Run `npm start` to launch the development server. To build an optimized bundle, run `npm run build`.

knightburton commented on src/app/app.scss:
```diff
@@ -0,0 +1,918 @@
+#info-panels .hidden-panel {
+    display: none;
+}
+
```
The watch panel css code is missing from here.

knightburton commented on src/app/client-protocol.js:
```diff
@@ -0,0 +1,921 @@
+    return returnValue;
+  }
+
+  nearest_offset = -1;
```
Add let or var (I prefer let since we using ES6 now) to declare the variable.

knightburton commented on src/app/client-protocol.js:
```diff
@@ -0,0 +1,921 @@
+ * @param {boolean} pending True if this breakpoint can be a pending breakpoint, false if not.
+ */
+DebuggerClient.prototype.setBreakpoint = function(str, pending) {
+  line = /^(.+):([1-9][0-9]*)$/.exec(str);
```
Add let or var to declare the "line" variable.

knightburton commented on src/app/client-protocol.js:
```diff
@@ -0,0 +1,921 @@
+ * @param {integer} index Index's of the breakpoint.
+ */
+DebuggerClient.prototype.deleteBreakpoint = function(index) {
+  breakpoint = this._activeBreakpoints[index];
```
Same, add let or var to declare the breakpoint.

knightburton commented on src/app/main.js:
```diff
@@ -0,0 +1,1025 @@
+  
+      session.addCommandToList(command);
+      session.setCommandCounter(session.getCommandList().length);
+      args = /^([a-zA-Z]+)(?:\s+([^\s].*)|)$/.exec(command);
```
Add let or var to declare the variable.

knightburton commented on src/app/memory-chart.js:
```diff
@@ -0,0 +1,558 @@
+   */
+  MemoryChart.prototype.resetChart = function() {
+    xAxisData = undefined;
+    estimatedTime = undefined;
```
This is not decared, this should be added to the list at the top of the file.

knightburton commented on src/index.html:
```diff
@@ -0,0 +1,455 @@
+        <i class="fa fa-dot-circle-o" aria-hidden="true" title="Turn on/off the Breakpoints panel."></i>
+        <div>Breakpoints</div>
+      </li>
+      <li class="sidenav-panel-toggle" id="sidenav-toggle-chart" data-pid="chart">
```
The watch panel toggle button code snippet is missing from here.

knightburton commented on src/index.html:
```diff
@@ -0,0 +1,455 @@
+              </div>
+            </div>
+          </div>
+          <!-- Memory usage -->
```
The watch panel code is missing from here.

daeyeon commented:
Thank you for the review. I missed css and html when updating #8 code base and they're now updated. Regarding implicit globals you pointed, I fixed them with `let`. For consistency, we need to consider defining a lint rule like https://eslint.org/docs/rules/no-var and removing all the `var` from the entire codes in another commit.

knightburton commented:
LGTM

knightburton commented on src/app/main.js:
```diff
@@ -0,0 +1,1040 @@
+  */
+  var debuggerObj = null;
+  
+    /**
```
The code indentation from here is too much.

daeyeon replied:
done

knightburton commented:
@daeyeon Please rebase this to the master again, two PR are landed.

daeyeon commented:
@knightburton The codes are updated with the last commit landed, #11.

knightburton commented:
@daeyeon nice, thank you!

yichoi commented:
LGTM
